src/core/components/parameter-row.jsx

```diff
@@ -63,24 +65,30 @@ export default class ParameterRow extends Component {
   }
 
   setDefaultValue = () => {
-    let { specSelectors, pathMethod, param } = this.props
+    let { specSelectors, pathMethod, rawParam } = this.props
 
-    if (param.get("value") !== undefined) {
-      return
-    }
+    let paramWithMeta = specSelectors.parameterWithMetaByIdentity(pathMethod, rawParam)
 
-    let schema = specSelectors.isOAS3() ? param.get("schema", Map({})) : param
 
-    let defaultValue = schema.get("default")
-    let xExampleValue = param.get("x-example") // Swagger 2 only
-    let parameter = specSelectors.parameterWithMetaByIdentity(pathMethod, param)
-    let value = parameter ? parameter.get("value") : ""
+    if (paramWithMeta.get("value") !== undefined) {
+      return
+    }
 
-    if( param.get("in") !== "body" ) {
-      if ( xExampleValue !== undefined && value === undefined && specSelectors.isSwagger2() ) {
-        this.onChangeWrapper(xExampleValue)
-      } else if ( defaultValue !== undefined && value === undefined ) {
-        this.onChangeWrapper(defaultValue)
+    if( paramWithMeta.get("in") !== "body" ) {
+      let newValue
+
+      if (specSelectors.isSwagger2()) {
+        newValue = paramWithMeta.get("x-example")
+          || paramWithMeta.getIn(["default"])
+          || paramWithMeta.getIn(["schema", "example"])
+          || paramWithMeta.getIn(["schema", "default"])
+      } else if (specSelectors.isOAS3()) {
+        newValue = paramWithMeta.get("example")
+          || paramWithMeta.getIn(["schema", "example"])
+          || paramWithMeta.getIn(["schema", "default"])
+      }
+      if(newValue !== undefined) {
+        this.onChangeWrapper(numberToString(newValue))
```

Comment:
curious: what happens if new value is `undefined` 🤔

Reply:
Everything not in the kongComponents folder is core swagger-ui code - and isn't actually used in our layout. We can start changing their master code if we want or you can bring it up on the master swagger-ui repo